onyo/lib/onyo.py

```diff
@@ -189,6 +196,7 @@ def restore(self) -> None:
         """
         self._git(['restore', '--source=HEAD', '--staged', '--worktree'] +
                   [str(file) for file in self.files_staged])
+        self.clear_caches()
```

Comment:
Why does this method clear all of the caches?

Reply:
I think it gets used when the user response is "no" to reset directories of mkdir, it is used by set to restore content and file name changes, and when edit/add will be allowed to edit or add templates (I propose this in #306) it might want to be used for these, too.

Reply:
Perhaps the wrong place to have asked, but in general, this question should be applied to all uses of `clear_caches()`.

It's not clear to me why invoking `add` should invalidate the cache of `templates`.

Reply:
The onyo commands we use do just change once the state of the repository, and commit, for these the `clear_cache()` is as far as I know not immediately relevant.

I was thinking rather about somebody writing there own python script and using `restore()` and `add()`.
e.g. something like this might happen:

```
repo = Repo(".")
# here doing anything which leads to loading the template cache
template = Path(".onyo/templates/new_device")
template.write("RAM: 16GB")
repo.add(template)
repo.commit("added new template")
# here doing something which would need the new template being in the cache
```

alternatively, somebody might want to copy our existing code (especially  `edit` or `new`), and just add something to the end of it:
```
    # commit changes
    staged = sorted(repo.files_staged)
    if staged:
        if not args.quiet:
            print(repo._diff_changes())
        if args.yes or request_user_response("Save changes? No discards all changes. (y/n) "):
            repo.commit(repo.generate_commit_message(message=args.message,
                                                     cmd="edit"))
        else:
            repo.restore()
            if not args.quiet:
                print('No assets updated.')

# here some code that needs the cache, after the user responded with "no"
```

I know these cases are currently not super relevant, because we do not have any commands doing something with the caches after commiting/restoring, and we have very weak template support (I mentioned this in #306 and still think this should be worked on).
But it feels like not updating the caches of the repository, after the state of the repository does change leads the `repo` var in an inaccurate state, after the user explicitely stated that something should be added or the whole thing should be restored.